internal/pkg/controllers/spod/configmap_controller.go

```diff
@@ -110,6 +110,17 @@ func (r *ReconcileSPOd) getConfiguredSPOd(cfg *corev1.ConfigMap) *appsv1.DaemonS
 			"--with-selinux=true")
 	}
 
+	enableLogEnricher, err := strconv.ParseBool(cfg.Data[config.SPOcEnableLogEnricher])
+	if err == nil && enableLogEnricher {
+		newSPOd.Spec.Template.Spec.Containers = append(
+			newSPOd.Spec.Template.Spec.Containers,
+			r.baseSPOd.Spec.Template.Spec.Containers[2])
+	}
+
+	// HostPID is only required for the log-enricher
+	// and is used to access cgroup files to map Process IDs to Pod IDs
+	newSPOd.Spec.Template.Spec.HostPID = (err == nil && enableLogEnricher)
```

Comment:
We could reduce the attack surface by putting it into a different daemonset, which probably would not make things easier to maintain.

Reply:
Yep, I was a bit torn on that. I am inclined to keep it for the time being - after all it is an optional component like selinuxd that currently runs privileged. And later review the overall attack surface and privileges. 

If we strive to have accurate security profiles per container we should be able to limit damage/risk of such privileges.

Reply:
I think this line could be moved up into the above `if` block and just set the value to `true`

Reply:
@cmurphy agreed, it does make it clearer.